client/my-sites/stats/hooks/use-stats-navigation.ts

```diff
@@ -0,0 +1,165 @@
+	store: '/stats/store/',
+	ads: '/stats/ads/',
+	subscribers: '/stats/subscribers/',
+	posts: '/stats/day/posts/',
+	authors: '/stats/day/authors/',
+	postDetails: null, // Last item in the history, the text is not displayed anywhere but this is used to track the item in history stack.
+};
+
+const SUPPORTED_QUERY_PARAMS: string[] = [
+	'startDate',
+	'endDate',
+	'num',
+	'summarize',
+	'chartStart',
+	'chartEnd',
+	'shortcut',
+];
+
+const defaultLastScreen = 'traffic';
+
+const getFilteredQueryParams = ( queryParams: QueryArgs ): QueryArgs => {
+	return Object.fromEntries(
+		Object.entries( queryParams ).filter( ( [ key ] ) => SUPPORTED_QUERY_PARAMS.includes( key ) )
+	);
+};
+
+/**
+ * Hook for managing stats navigation state
+ * Supports reading/writing from sessionStorage and initializing from query params
+ * @returns { { text: string; url: string | null } }
+ */
+export const useStatsNavigationHistory = (): { text: string; url: string | null } => {
+	const [ lastScreen, setLastScreen ] = useState< {
+		lastScreen: string;
+		queryParams: QueryArgs;
+	} >( {
+		lastScreen: defaultLastScreen,
+		queryParams: {},
+	} );
+	const siteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+
+	useEffect( () => {
+		try {
+			const navState = JSON.parse( sessionStorage.getItem( STORAGE_KEY ) || '[]' );
+
+			// Select the second last item from the history stack as the back link.
+			// The last item in the stack if the current screen.
+			const lastItem =
+				Array.isArray( navState ) && navState.length >= 2 ? navState[ navState.length - 2 ] : {};
+
+			// Make sure it's array and select last item
+			if ( lastItem && lastItem.lastScreen ) {
+				setLastScreen( lastItem );
+			} else {
+				setLastScreen( {
+					lastScreen: defaultLastScreen,
+					queryParams: {},
+				} );
+			}
+		} catch ( e ) {}
+	}, [] );
+
+	const backLink = useMemo( () => {
+		if ( ! siteSlug ) {
+			return null;
+		}
+
+		const backLink = possibleBackLinks[ lastScreen.lastScreen ];
+
+		if ( ! backLink ) {
+			return null;
+		}
+
+		const queryParams = buildQueryString( getFilteredQueryParams( lastScreen.queryParams ) );
+
+		return backLink + siteSlug + ( queryParams ? '?' + queryParams : '' );
+	}, [ lastScreen, siteSlug ] );
+
+	return {
+		text: localizedTabNames[ lastScreen.lastScreen ] || '',
+		url: backLink,
+	};
+};
+
+/**
+ * Utility to record the current screen for back navigation
+ * @param {string} screen - Current screen identifier
+ * @param {Object} queryParams - Query parameters for the screen
+ * @param {boolean} reset - Whether to reset the navigation history
+ */
+export const recordCurrentScreen = (
+	screen: string,
+	queryParams: QueryArgs = {},
+	reset: boolean = false
+): void => {
+	try {
+		if ( ! screen || ! ( screen in localizedTabNames ) ) {
+			return;
+		}
+
+		const filteredQueryParams = getFilteredQueryParams( queryParams );
+		const currentEntry = {
+			lastScreen: screen,
```

Comment:
Is this actually about the current screen, rather than the _last screen_?
```suggestion
			screen,
```

Reply:
Agreed, Done in faf6bfd91548028f32ea4c41ae7bec981f378650